Python/generated_cases.c.h

```diff
@@ -1745,11 +1745,13 @@
 
         TARGET(LOAD_ATTR) {
             PREDICTED(LOAD_ATTR);
+            PyObject *owner = PEEK(1);
+            PyObject *res;
+            PyObject *res2;
```

Comment:
Should this initialize `res2 = NULL`? Right now I get compiler warnings (in GitHub) about a possibly uninitialized variable:
```
‘res2’ may be used uninitialized in this function [-Wmaybe-uninitialized]
```
presumably from the `PUSH()` on L1798.

Reply:
I think the issue is that the compiler can't see that ``if (oparg & 1)`` is the same as ``if (_cond_res2)``, so it can't tell that res2 is always initialised when it is used.  Not sure what to do about it, add a syntax in the DSL to access the condition from within the bytecode definition?

Reply:
I'll just initialize it to NULL. Perhaps some later phase of optimization will recognize that for the dead value it is.

Reply:
I think if we rename `_cond_res2` to `res2_exists` and then in the body of the bytecode we do `if(res2_exists) /* use res2 */` instead of `if(oparg &1) /* use res2 */` then it would make the code clearer.  It's also safer, because if you change the args to the instr then `oparg&1` might not be correct anymore, but `res2_exists` will be automatically updated.

Reply:
Okay, let's try that. I had a commit in the queue that initializes the variable to NULL, which I think would also get rid of the problem (if not I'd be very surprised) but since I didn't push that let's first try it with your idea.

Reply:
Hm, looks like it doesn't help, the error is still there. So instead let me try this with the (much simpler) "= NULL" solution, which *has* to work. In fact, I may also drop the flag variable idea, and solve the warning about bool by writing `X ? 1 : 0` instead of `X != 0`.